templates/home.html

```diff
@@ -13,7 +13,82 @@ <h2>Home</h2>
     <h2 class="sub_title">{{ tweet.content }}</h2>
     <small>{{ tweet.created_at }}{{tweet.user.username}}</small>
     <a href="{% url 'tweets:detail' tweet.pk %}">Check it out</a>
+    {% if tweet.pk in like_list %}
+    <button class="like_button" id="js-like-for-tweet" style="border:none;background:none" data-tweet_pk="{{tweet.pk}}"
+        data-is_liked="true">
+        <i class="fas fa-heart text-danger" id="js-like-for-tweet-icon-{{tweet.pk}}"></i>
+    </button>
+    {% else %}
+    <button class="like_button" id="js-like-for-tweet" style="border:none;background:none" data-tweet_pk="{{tweet.pk}}"
+        data-is_liked="false">
+        <i class="far fa-heart text-danger" id="js-like-for-tweet-icon-{{tweet.pk}}"></i>
+    </button>
+    {% endif %}
+    <span id="js-like-for-tweet-count-{{tweet.pk}}">{{ tweet.like_set.count }}</span>
+    <span>people liked it !!</span>
 </div>
+
 {% endfor %}
+{% endblock %}
+
+{% block extrajs %}
+<script type="text/javascript">
+    function getCookie(name) {
+        let cookieValue = null;
+        if (document.cookie && document.cookie !== '') {
+            const cookies = document.cookie.split(';');
+            for (let i = 0; i < cookies.length; i++) {
+                const cookie = cookies[i].trim();
+                if (cookie.substring(0, name.length + 1) === (name + '=')) {
+                    cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+                    break;
+                }
+            }
+        }
+        return cookieValue;
+    }
+    const csrftoken = getCookie('csrftoken');
+
+    const $likeButtons = document.getElementsByClassName("like_button")
+    for (const $likeButton of $likeButtons) {
+        const $is_liked = $likeButton.dataset.is_liked

```

Comment:
```suggestion
        let is_liked = likeButton.dataset.is_liked
```
このようにconstではなくletにしないと再代入が行えなくなり、true, falseの値を代入して切り替えるという操作ができなくなるので気を付けてください。

Reply:
確かに！

Reply:
e.currentTargetで要素取ってきていいね複数回押せるようにしたからこの変数消したけどええか？

Reply:
e.currentTargetで取ってきても良いのですが、無駄な記述になってる気がします。
e.currentTarget == likeButton は true すなわち同じものです。これを今まで is_liked = likeButton.dataset.is_liked としていたのは、毎回 datasetまで書く記述を減らすために変数に入れていたと思ってたのですが違いました？

Reply:
そのつもりだったんだけどconstをletに変えても再代入できないっぽいので代替としてe.currentTargetを使うことにした

Reply:
嘘ですもう一回やったらちゃんと通りましたごめんなさい